File: src/test/java/it/xpug/kata/birthday_greetings/EmployeeTest.java
package it.xpug.kata.birthday_greetings;
import static org.junit.Assert.*;

import org.junit.*;



public class EmployeeTest {

	@Test
	public void testBirthday() throws Exception {
		Employee employee = new Employee("foo", "bar", "a@b.c", new XDate("1990/01/31"));
		assertFalse("not his birthday", employee.isBirthday(new XDate("2008/01/30")));
		assertTrue("his birthday", employee.isBirthday(new XDate("2008/01/31")));
	}

	@Test
	public void equality() throws Exception {
		Employee base = new Employee("First", "Last", "first@last.com", new XDate("1999/09/01"));
		Employee same = new Employee("First", "Last", "first@last.com", new XDate("1999/09/01"));
		Employee different = new Employee("First", "Last", "boom@boom.com", new XDate("1999/09/01"));

		assertFalse(base.equals(null));
		assertFalse(base.equals(""));
		assertTrue(base.equals(same));
		assertFalse(base.equals(different));
	}
}

File: src/test/kotlin/it/xpug/kata/birthday_greetings/EmployeeTest.kt
package it.xpug.kata.birthday_greetings

import io.kotlintest.matchers.shouldBe


import io.kotlintest.specs.StringSpec

class EmployeeTest : StringSpec({
    "employee has a birthday" {
        val employee = Employee("foo", "bar", XDate("1990/01/31"), "a@b.c")
        employee.isBirthday(XDate("2008/01/30")) shouldBe false
        employee.isBirthday(XDate("2008/01/31")) shouldBe true
    }
    "employee equality" {
        val base = Employee("First", "Last", XDate("1999/09/01"), "first@last.com")
        val same = Employee("First", "Last", XDate("1999/09/01"), "first@last.com")
        val different = Employee("First", "Last", XDate("1999/09/01"), "boom@boom.com")

        (base == same) shouldBe true
        (base === same) shouldBe false
        (base == different) shouldBe false
    }
})